docs/howto/logging.txt

```diff
@@ -0,0 +1,339 @@
+
+    LOGGING = {
+        [...]
+        'handlers': {
+            'file': {
+                'class': 'logging.FileHandler',
+                'filename': 'general.log',
+            },
+        },
+    }
+
+Different handler classes take different configuration options. For more
+information on available handler classes, see the
+:class:`~django.utils.log.AdminEmailHandler` provided by Django and the various
+:py:mod:`handler classes <logging.handlers>` provided by Python.
+
+Logging levels can also be set on the handlers (by default, they accept log
+messages of all levels). Using the example above, adding:
+
+..  code-block:: python
+    :emphasize-lines: 4
+
+    {
+        'class': 'logging.FileHandler',
+        'filename': 'general.log',
+        'level': 'DEBUG',
+    }
+
+would define a handler configuration that only accepts records of level
+``DEBUG`` and higher.
+
+Configure a logger mapping
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To send records to this handler, configure a logger mapping to use it for
+example:
+
+..  code-block:: python
+    :emphasize-lines: 3-8
+
+    LOGGING = {
+        [...]
+        'loggers': {
+            '': {
+                'level': 'DEBUG',
+                'handlers': ['file'],
+            },
+        },
+    }
+
+The mapping's name determines which log records it will process. This
+configuration (``''``) is *unnamed*. That means that it will process records
+from *all* loggers (see :ref:`naming-loggers` below on how to use the mapping
+name to determine the loggers for which it will process records).
+
+It will forward messages of levels ``DEBUG`` and higher to the handler named
+``file``.
+
+Note that a logger can forward messages to multiple handlers, so
+the relation between loggers and handlers is many-to-many.
+
+If you execute::
+
+    logger.debug('Attempting to connect to API')
+
+in your code, you will find that message in the file ``general.log`` in the
+root of the project.
+
+Configure a formatter
+~~~~~~~~~~~~~~~~~~~~~
+
+By default, the final log output contains the message part of each
+:class:`log record <logging.LogRecord>`. Use a formatter if you want to
+include additional data. First define your formatters, for example:
+
+..  code-block:: python
+    :emphasize-lines: 3-12
+
+    LOGGING = {
+        [...]
+        'formatters': {
+            'verbose': {
+                'format': '{name} {levelname} {asctime} {module} {process:d} {thread:d} {message}',
+                'style': '{',
+            },
+            'simple': {
+                'format': '{levelname} {message}',
+                'style': '{',
+            },
+        },
+    }
+
+(The ``style`` keyword allows you to specify ``{`` for :meth:`str.format` or
+``$`` for :class:`string.Template` formatting; the default is ``$``.)
+
+See :ref:`logrecord-attributes` for the :class:`~logging.LogRecord` attributes
+you can include.
+
+To apply a formatter to a handler, configure the handler, for example:
```

Comment:
It's implicitly suggesting that the `'formatter'` value is the `key` in the `formatters` dictionary (ie: the name of the formatter is the key), but it could be explicitly stated. 

Because at this point one has just declared `formatters.verbose` and now `formatter.verbose` referring to it and that's an unfortunate necessity but could be confusing.